Lib/socket.py

```diff
@@ -203,10 +205,12 @@ def accept(self):
         For IP sockets, the address info is a pair (hostaddr, port).
         """
         fd, addr = self._accept()
-        # If our type has the SOCK_NONBLOCK flag, we shouldn't pass it onto the
-        # new socket. We do not currently allow passing SOCK_NONBLOCK to
-        # accept4, so the returned socket is always blocking.
-        type = self.type & ~globals().get("SOCK_NONBLOCK", 0)
+        type = self.type
+        if _SOCK_NONBLOCK:
```

Comment:
Either `_SOCK_NONBLOCK` should be `None` if not present in `socket` -- then `if _SOCK_NONBLOCK` will be very fast check.
Or drop `if`: `type & ~0` works pretty well

Reply:
I agree with Andrew, I don't think that the if is needed.